OSVROpenGL/app/src/main/jni/main.cpp

```diff
@@ -652,6 +626,12 @@ void stop() {
         delete gClientContext;
         gClientContext = nullptr;
     }
+
+    if(gServer) {
+        LOGI("[OSVR] Shutting down server...");
+        gServer->stop();
```

Comment:
According to the docs, the ```stop``` method is supposed to signal server stop and then wait for it to happen. There is a separate ```signalStop``` that is meant to just set the flag and not wait. From the implementation in ServerImpl.cpp, it appears to be waiting.